controller/cache/cache.go

```diff
@@ -69,6 +70,11 @@ const (
 
 	// EnvClusterCacheRetryUseBackoff is the env variable to control whether to use a backoff strategy with the retry during cluster cache sync
 	EnvClusterCacheRetryUseBackoff = "ARGOCD_CLUSTER_CACHE_RETRY_USE_BACKOFF"
+
+	// AnnotationApplyResourcesUpdate when set to true on a resource that is not tracked under an app, argocd will generate a
+	// hash and apply `ignoreResourceUpdate` configuration on it. If the annotation is set to false (or not presented) on a resource
+	// that is not tracked under an app, ignoreResourceUpdates configuration will not be applied.
```

Comment:
Leaking implementation details on the API.

```suggestion
	// AnnotationApplyResourcesUpdate when set to true on an untracked resource, 
	// argo will apply `ignoreResourceUpdates` configuration on it.
```